test/Microsoft.ML.Tests/TrainerEstimators/CalibratorEstimators.cs

```diff
@@ -0,0 +1,149 @@
+
+            // fixedPlatCalibrator
+            var fixedPlatCalibratorEstimator = new FixedPlattCalibratorEstimator(Env, calibratorTestData.transformer.Model, labelColumn: "Label", featureColumn: "Features");
+            var fixedPlatCalibratorTransformer = fixedPlatCalibratorEstimator.Fit(calibratorTestData.scoredData);
+
+            checkValidCalibratedData(calibratorTestData.scoredData, fixedPlatCalibratorTransformer);
+
+            //test estimator
+            TestEstimatorCore(calibratorTestData.pipeline, calibratorTestData.data);
+
+            Done();
+        }
+
+        /// <summary>
+        /// OVA and calibrators
+        /// </summary>
+        [Fact]
+        public void NaiveCalibratorEstimator()
+        {
+            var calibratorTestData = GetCalibratorTestData();
+
+            // naive calibrator
+            var naiveCalibratorEstimator = new NaiveCalibratorEstimator(Env, calibratorTestData.transformer.Model, "Label", "Features");
+            var naiveCalibratorTransformer = naiveCalibratorEstimator.Fit(calibratorTestData.scoredData);
+
+            // check data
+            checkValidCalibratedData(calibratorTestData.scoredData, naiveCalibratorTransformer);
+
+            //test estimator
+            TestEstimatorCore(calibratorTestData.pipeline, calibratorTestData.data);
+
+            Done();
+        }
+        /// <summary>
+        /// OVA and calibrators
+        /// </summary>
+        [Fact]
+        public void PavCalibratorEstimator()
+        {
+            var calibratorTestData = GetCalibratorTestData();
+
+            // pav calibrator
+            var pavCalibratorEstimator = new PavCalibratorEstimator(Env, calibratorTestData.transformer.Model, "Label", "Features");
+            var pavCalibratorTransformer = pavCalibratorEstimator.Fit(calibratorTestData.scoredData);
+
+            //check data
+            checkValidCalibratedData(calibratorTestData.scoredData, pavCalibratorTransformer);
+
+            //test estimator
+            TestEstimatorCore(calibratorTestData.pipeline, calibratorTestData.data);
+
+            Done();
+        }
+
+        CalibratorTestData GetCalibratorTestData()
+        {
+            var (pipeline, data) = GetBinaryClassificationPipeline();
+            var binaryTrainer = new AveragedPerceptronTrainer(Env);
+
+            pipeline = pipeline.Append(binaryTrainer);
+
+            var transformer = pipeline.Fit(data);
+            var scoredData = transformer.Transform(data);
+            var scoredDataPreview = scoredData.Preview();
+            Assert.True(scoredDataPreview.ColumnView.Length == 5);
+
+            return new CalibratorTestData
+            {
+                data = data,
+                scoredData = scoredData,
+                pipeline = pipeline,
+                transformer = ((TransformerChain<BinaryPredictionTransformer<LinearBinaryPredictor>>)transformer).LastTransformer as BinaryPredictionTransformer<LinearBinaryPredictor>,
+            };
```

Comment:
Do you wan to test linear regression here?

Reply:
no, it would be overkill. As longs as the columns are produced, if the score column is not there, it will faile later with that message. 

---
In reply to: [240346246](https://github.com/dotnet/machinelearning/pull/1816#discussion_r240346246) [](ancestors = 240346246)